tasks/main.yml

```diff
@@ -90,17 +103,20 @@
   set_fact:
     __vpn_psks: "{{ __vpn_psk_hostvars.__vpn_psks }}"
 
-- name: build tunnels
+- name: build host-to-host tunnels
+  when: not (conn.opportunistic | d(vpn_opportunistic))
   vars:
     tunnels: |
       {% set unique_tunnels = [] %}
       {% for tunnel in vpn_connections %}
-      {%   if inventory_hostname in tunnel.hosts %}
-      {%     for node in tunnel.hosts %}
-      {%       if node != inventory_hostname %}
-      {%         set _ = unique_tunnels.append(node) %}
-      {%       endif %}
-      {%     endfor %}
+      {%   if not tunnel.opportunistic | d(vpn_opportunistic) %}
```

Comment:
do you need to check for this condition again, since the `when` condition should already make the condition `false` here?

Reply:
There were some strange things happening with the conditional statements in this block and definitely some redundancy. It seems like the outer block conditional was not stopping the tasks from running anyway (even when the configuration was opportunistic). I ended up keeping this `if` statement and removing the outer one. It seems to do the job, though I'm still a little puzzled about why the outer conditional was not applied to the whole block of tasks

Reply:
I know that if you use `when` in a `loop`, then the `when` applies to each iteration of the loop - but I didn't think that was being used here

Reply:
Yeah, this situation is a bit different from a `loop`. I was referring to the `when` that was attached to the whole `build host-to-host tunnels` block (not sure if we were talking about the same thing). So I just expected all three tasks to be skipped if the `when not (tunnel.opportunistic | d(vpn_opportunistic))` condition was false, but that didn't seem to be what was happening